zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -13,31 +13,233 @@ enum class Status {
+
+    when {
+        travelDistanceCars == null -> ValidationResult(Status.MISSING_DATA, "Cars travel distance is not provided")
+        travelDistanceCars in 5000..100000 -> ValidationResult(Status.VALID, "Cars travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Cars travel distance are outside the valid range (5k..100k km)")
+    }
+}
+val validateTruckTravelDistance = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    val travelDistanceTrucks = gridConnection.transport.trucks.annualTravelDistancePerTruckKm
+
+    when {
+        travelDistanceTrucks == null -> ValidationResult(Status.MISSING_DATA, "Trucks travel distance is not provided")
+        travelDistanceTrucks in 5000..100000 -> ValidationResult(Status.VALID, "Trucks travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Trucks travel distances are outside the valid range (5k..100k km)")
+    }
+}
+val validateVanTravelDistance = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    val travelDistanceVans = gridConnection.transport.vans.annualTravelDistancePerVanKm
+
+    when {
+        travelDistanceVans == null -> ValidationResult(Status.MISSING_DATA, "Vans travel distance is not provided")
+        travelDistanceVans in 5000..100000 -> ValidationResult(Status.VALID, "Vans travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Vans travel distances are outside the valid range (5k..100k km)")
+    }
+}
+
+
+// Validator for number of electric vehicles should be less than or equal to total number of vehicles
+val validateTotalElectricCars = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    when {
+        (gridConnection.transport.cars.numElectricCars ?: 0) > (gridConnection.transport.cars.numCars ?: 0) -> ValidationResult(Status.INVALID, "Number of electric cars exceeds the total number of cars")
+        else -> ValidationResult(Status.VALID, "Number of Electric Cars are valid")
+    }
+}
+
+val validateTotalElectricTrucks = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    when {
+        (gridConnection.transport.trucks.numElectricTrucks ?: 0) > (gridConnection.transport.trucks.numTrucks ?: 0) -> ValidationResult(Status.INVALID, "Number of electric trucks exceeds the total number of trucks")
+        else -> ValidationResult(Status.VALID, "Number of Electric Trucks are valid")
+    }
+}
+
+
+val validateTotalElectricVans = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    when {
+        (gridConnection.transport.vans.numElectricVans ?: 0) > (gridConnection.transport.vans.numVans ?: 0) -> ValidationResult(Status.INVALID, "Number of electric vans exceeds the total number of vans")
+        else -> ValidationResult(Status.VALID, "Number of Electric Vans are valid")
```

Comment:
Nit: we don't know if the number is valid, we just know it's consistent with the total.